source/code-snippets/crud/arrayFilters.js

```diff
@@ -6,202 +6,130 @@ const uri =
 
-    console.log(JSON.stringify(await (await pizza.find()).toArray()));
+    console.log(JSON.stringify(await (await myColl.find()).toArray()));
 
     // start firstArrayElement example
-    const query = { name: "Steve Lobsters", "items.type": "pizza" };
+    const query = { name: "Sandy Kane", "appointments.stylist": "Janna" };
     const updateDocument = {
-      $set: { "items.$.size": "extra large" }
+      $push: { "appointments.$.services": "wash" },
     };
-    const result = await pizza.updateOne(query, updateDocument);
+    const result = await myColl.updateOne(query, updateDocument);
     // end firstArrayElement example
     console.log(result.modifiedCount);
-    console.log(JSON.stringify(await (await pizza.find()).toArray()));
+    console.log(JSON.stringify(await (await myColl.find()).toArray()));
   } finally {
     await client.close();
   }
 }
 
-
-async function arrayFiltersOne() {
+async function runAllArrayElements() {
   try {
     const database = client.db("test");
-    const pizza = database.collection("pizza");
+    const myColl = database.collection("salonClients");
 
-    console.log(JSON.stringify(await (await pizza.find()).toArray()));
+    console.log(JSON.stringify(await (await myColl.find()).toArray()));
 
-    // start arrayFiltersOne example
-    const query = { name: "Steve Lobsters" };
+    // start allArrayElement example
+    const query = { name: "Dennis Roberts" };
     const updateDocument = {
-      $push: { "items.$[orderItem].toppings": "garlic" }
-    };
-    const options = {
-      arrayFilters: [{
-        "orderItem.type": "pizza",
-        "orderItem.size": "large",
-      }]
+      $inc: { "appointments.$[].cost": -15 },
     };
-
-    const result = await pizza.updateMany(query, updateDocument, options);
-    // end arrayFiltersOne example
-
+    const result = await myColl.updateOne(query, updateDocument);
+    // end allArrayElement example
     console.log(result.modifiedCount);
-    console.log(JSON.stringify(await (await pizza.find()).toArray()));
+    console.log(JSON.stringify(await (await myColl.find()).toArray()));
   } finally {
     await client.close();
   }
 }
 
-async function arrayFiltersTwo() {
+async function arrayFiltersIdentifier() {
   try {
     const database = client.db("test");
-    const pizza = database.collection("pizza");
+    const myColl = database.collection("salonClients");
 
-    console.log(JSON.stringify(await (await pizza.find()).toArray()));
+    console.log(JSON.stringify(await (await myColl.find()).toArray()));
 
-    // start arrayFiltersTwo example
-    const query = { name: "Steve Lobsters" };
+    // start arrayFiltersIdentifier example
+    const query = { name: "Sandy Kane" };
     const updateDocument = {
-      $push: { "items.$[item].toppings": "salami" },
+      $set: {
+        "appointments.$[appt].paymentPlan": {
+          payment1: 50,
+          payment2: 50,
+          payment3: "remaining balance",
```

Comment:
Question: is it necessary to use three fields to demonstrate this example? Or could it be limited to one for brevity? E.g. "{ planType: "monthly installment"} "?

Reply:
I thought it could be useful to show how to set an embedded document. LMK if you still think this should be shortened

Reply:
I thought you meant that the new field could just be a string field. I'm fine changing the embedded document to be one field!